devito/data/data.py

```diff
@@ -132,16 +140,76 @@ def _global(self, glb_idx, decomposition):
         ret._is_distributed = any(i is not None for i in decomposition)
         return ret
 
+    def check_slicing(func):
+        """Check if the operation will communication across MPI ranks."""
```

Comment:
will trigger communication?

instead of "operation" you can be more specific -- ``__getitem__`` or ``__setitem__`` right?

Reply:
> will trigger communication?
Yes, this will trigger at least some communication.